### PR TITLE
! s and u tags could not be disabled in editor + construct is tidier

### PR DESCRIPTION
**Pull request by Underdog-01**
- changed if/else to array & foreach construct
- if/else had incorrect conditions anyhow since "i" was repeated in error
  (more to come with this file since bbc button array is not working properly atm)

Signed-off-by: -Underdog- github.underdog@gmail.com

### CODE REVIEW

**Comment by AngelinaBelle** [comment]
This fixes the bug of s and u tags not being dealt with properly, and looks like easier-to-maintain code.

File: Sources/Subs-Editor.php [hunk 1]
```diff
@@ -1650,6 +1650,18 @@ function create_control_richedit($editorOptions)
...
+			'img' => 'image',
+			'url' => 'link',
+			'sup' => 'superscript',
+			'sub' => 'subscript',
+			'hr' => 'horizontalrule',
+		);
+
 		// Allow mods to modify BBC buttons.
 		// Note: pass the array here is not necessary and is deprecated, but it is ketp for backward compatibility with 2.0
 		call_integration_hook('integrate_bbc_buttons', array(&$context['bbc_tags']));
```
Comment (live627) [inline Sources/Subs-Editor.php:1667]: Maybe pass $disabled_AltTags into the hook? Rename it to $disabled_editor_tags?

**Comment by Underdog-01** [comment]
I can change the name of the array used if you think it is more suitable.
I can also add the array to the hook call but I don;t see the point atm since the hooked BBC's show up on the BBC admin page whereas they can be unchecked. The loop that follows should disable them anyway.

Signed-off-by: -Underdog- github.underdog@gmail.com

**Comment by live627** [comment]
his pull request contains a merge conflict now due to your other PR

**Comment by Underdog-01** [comment]
I edited the code to match the other PR.
If it still gives a conflict let me know & I will close this one and start another without the last part.

Signed-off-by: -Underdog- github.underdog@gmail.com

**Comment by live627** [comment]
Look for the green icon below this.

**Comment by XinYenFon** [comment]
you didn't sign-off your last two commit :(

**Comment by Underdog-01** [comment]
It won't let me edit the actual title relating to the code edit.
I will just close this one and do it again.

**Comment by Underdog-01** [comment]
keep in mind that I will be posting more PR's for this file after I take the liberty to fix the WYSIWYG editor. the bbc array needs to include the html from the bbc tag (added via hook or otherwise) to have that behave properly but the current code for the WYSIWYG function needs to be reworked first. 
Atm its buggy in SMF 2.1 as is the behavior for SMF 2.0.8... I will look at it when time permits.
Any hooked bbc tags that do not include the html in the array will behave the same... mod authors will have to opt in html to the array to have it behave different for the WYSIWYG editor .. if you follow what I am getting at?

**Pull request closed without merging**